src/material/core/private/ripple-loader.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {
+  ANIMATION_MODULE_TYPE,
+  ElementRef,
+  Injectable,
+  NgZone,
+  OnDestroy,
+  inject,
+} from '@angular/core';
+import {MAT_RIPPLE_GLOBAL_OPTIONS, MatRipple} from '../ripple';
+import {Platform} from '@angular/cdk/platform';
+
+/** The options for the MatRippleLoader's event listeners. */
+const eventListenerOptions = {capture: true};
+
+/** The events that should trigger the initialization of the ripple. */
+const rippleInteractionEvents = ['focus', 'click', 'mouseenter', 'touchstart'];
+
+/** The attribute attached to a component whose ripple has not yet been initialized. */
+const matRippleUninitialized = 'mat-ripple-uninitialized';
+
+/** Additional classes that should be added to the ripple when it is rendered. */
+const matRippleClassName = 'mat-ripple-class-name';
+
+/** Whether the ripple should be centered. */
+const matRippleCentered = 'mat-ripple-centered';
+
+/** Whether the ripple should be disabled. */
+const matRippleDisabled = 'mat-ripple-disabled';
+
+/**
+ * Handles attaching ripples on demand.
+ *
+ * This service allows us to avoid eagerly creating & attaching MatRipples.
+ * It works by creating & attaching a ripple only when a component is first interacted with.
+ */
+@Injectable({providedIn: 'root'})
+export class MatRippleLoader implements OnDestroy {
+  private _document = inject(DOCUMENT, {optional: true});
+  private _animationMode = inject(ANIMATION_MODULE_TYPE, {optional: true});
+  private _globalRippleOptions = inject(MAT_RIPPLE_GLOBAL_OPTIONS, {optional: true});
+  private _platform = inject(Platform);
+  private _ngZone = inject(NgZone);
+
+  constructor() {
+    this._ngZone.runOutsideAngular(() => {
+      for (const event of rippleInteractionEvents) {
+        this._document?.addEventListener(event, this._onInteraction, eventListenerOptions);
```

Comment:
`_onInteraction` appears to be using `this`, so you need to bind it

Reply:
Currently, `_onInteraction` uses arrow functions to keep it's scope. Should I switch to using `bind` instead?

Reply:
Oh I see, my bad I didn't notice that